frame/babe/src/lib.rs

```diff
@@ -305,12 +305,31 @@ impl<T: Trait> Module<T> {
 		// epoch 0 as having started at the slot of block 1. We want to use
 		// the same randomness and validator set as signalled in the genesis,
 		// so we don't rotate the epoch.
-		now != sp_runtime::traits::One::one() && {
+		now != One::one() && {
 			let diff = CurrentSlot::get().saturating_sub(Self::current_epoch_start());
 			diff >= T::EpochDuration::get()
 		}
 	}
 
+	/// Return the _best guess_ block number, at which the next epoch change is predicted to happen.
+	///
+	/// In other word, this is only accurate if no slots are missed. Given missed slots, the slot
+	/// number will grow while the block number will not. Hence, the result can be interpreted as an
+	/// upper bound.
+	///
+	/// -------------- IMPORTANT NOTE --------------
+	/// This implementation is linked to how [`should_epoch_change`] is working. This might need to
```

Comment:
Fair. Although, let's . wait a bit more to see if we even need this. If the consensus is to allow solutions all the time or a few sessions ahead, then all of this can just go away.

Reply:
What I mean here was that both share the logic of slots and epochs. Otherwise, the code cannot really be shared, unless if we call `should_epoch_change()` in a loop until it returns true? that would be uncool.

Reply:
last call on this: the functions are literally placed next to each other. I don't think much else can be done for it. The phrase,  `IMPORTANT NOTE` is a bit like we're doing a foot-gun but no; the epoch change mechanism is really the core of the babe module. It will most likely never change.